dist_crash.erl

```diff
@@ -0,0 +1,27 @@
+-module(dist_crash).
+-export([main/1]).
+
+-mode(compile).
+
+main([]) ->
+    inet_db:set_lookup([file]),
```

Comment:
@max-au Do you remember if this `inet_db:set_lookup([file])` does anything useful for the test?

Reply:
Iit makes the connection attempt to fail fast. Otherwise DNS resolution kicks in, taking a long time, and since connections are all serialised through net_kernel, the race isn't triggered.

Side note, I think it's worth making hostname resolution async, to avoid head-of-the-line issue when a single delayed DNS request may trigger long wait time.